### ISSUE TITLE
Graphics HD remake

### ISSUE THREAD

**Issue by jan-kleks** [pjasicek/OpenClaw#124]
Maybe some crazy fan who is also an artist would be interested in remaking the graphics in HD? But it's a lot of work, probably one would have to paint over original assets... Imagine how cool Claw could look then on an HD screen. **Leave this open, please, maybe we will find someone**.

Of course, the original graphics could stay as an option.

Inspiration: [Day of the Tentacle Remastered gameplay and graphics comparison (1993 vs. 2016)](https://www.youtube.com/watch?v=KGjKBMvhaL0)

**Comment by jstasiak** [comment]
I'll just leave this here: https://github.com/alexjc/neural-enhance I'd be interested to see how well could neural-enhance (or some other tool) work with Captain Claw assets.

**Comment by pjasicek** [comment]
Hehe, that would be awesome.

**Comment by iisfitblud** [comment]
I would love to help remaster the graphics. Although I can't seem to find any image files of any sort? If anyone can help me get started as to what I need to change i'll happily create all new graphics.

**Comment by pjasicek** [comment]
That's great !

Here are converted images ing .PNG format:

https://drive.google.com/file/d/1l3ZgBw-HCG7CmlMrwvQQwJJQJ8FRVFq7/view

Thanks !
_____________________________

On a more technical side:

Currently, OpenClaw uses all of their original asset from original .REZ archive. The images in .REZ archive are stored in a proprietary .PID format (this format was only used in some of Monolith's games) and OpenClaw used .PID format directly. 

**The .PID format stores 2 key informations which no normal image format does: X and Y offset of any particular image in a spritesheet.**

What that means is that if the enhanced graphics would result in for example .PNG format, I would have to rewrite some parts of my image handling, but I would ultimately take care of that.

The only thing I would require is for the new images to be X times larger than original ones where X is 2, 3 or 4 (exactly) so that image offsets could also be applied and so that images in spritesheet animations would not result in images having different offsets.

**Comment by iisfitblud** [comment]
Thanks for this. Unfortunately I'm away for the weekend now, so won't get to play with things until Sunday. This is going to be hell of a job to do on my own, but happy to give it a go, I have a lot of experience with sprites etc.

I'm going to start with the captain then do level 1 and see how it goes.

I assume that you've not been contacted in regards to a cease and desist at all yet? It would kill me (and you I'm sure) if that were to happen after putting time in.

Also, happy to have a chat properly about it in discord at some point of you'd like too.

Thanks, Ryan.

---- Petr Jašíček wrote ----


That's great !

Here are converted images ing .PNG format:

https://drive.google.com/file/d/1l3ZgBw-HCG7CmlMrwvQQwJJQJ8FRVFq7/view

Thanks !

________________________________

On a more technical side:

Currently, OpenClaw uses all of their original asset from original .REZ archive. The images in .REZ archive are stored in a proprietary .PID format (this format was only used in some of Monolith's games) and OpenClaw used .PID format directly.

The .PID format stores 2 key informations which no normal image format does: X and Y offset of any particular image in a spritesheet.

What that means is that if the enhanced graphics would result in for example .PNG format, I would have to rewrite some parts of my image handling, but I would ultimately take care of that.

The only thing I would require is for the new images to be X times larger than original ones where X is 2, 3 or 4 (exactly) so that image offsets could also be applied and so that images in spritesheet animations would not result in images having different offsets.

**Comment by pjasicek** [comment]
No C&D or anything pointing towards it. This repository does not contain any resources from the original game whatsoever.

**Comment by iisfitblud** [comment]
Okay,

Attached is a Rar containing new assests for all of Claws frames.

Instead of redoing them, I thought I would first try using a scale algorithm on them to see what the difference was.


All the files are 2X bigger than the orginals.

If you could replace the assests and check it out yourself and also send me a copy so I can view it as well that would be great.

Let me know if you have any issues.

Ryan.


________________________________

**Comment by pjasicek** [comment]
Sorry, but I don't see any attached archive.

Regarding replacing only certain actor's with higher resolution ones - this would unfortunately not really work. Either all level's actors would have to be replaced or none of them.

 1) If I replaced just Claw's images, Claw would just be 2x bigger than everyone else. Keeping everyone else with the same old low-res sprites and Claw 2x bigger would mean upscaling everything else, which is not really an option.
2) Using PNG's instead of original PIDs would mean parts of my image and animation subsystems rewrite (described in my previous reply)

I remember myself also trying to enhance the images by using ON1 Resize's "Genuine Fractal" and "High Pass" filters. The results as far as I remember were not nearly good enough for me to go through the trouble.

On the DRM and licensing side this is also a violation unfortunately. Ripping, enhancing and using assets from original game is really a different thing than making them from scratch (good reference here is Open Transport Tycoon Deluxe's approach).

Petr

**Comment by iisfitblud** [comment]
No worries, I won't bother resending at the moment then.


I am happy to go through and do this for all the assets, it's just going to take some time.

I think it does look better for it, so I'm happy to put in the time.

I don't know how to convert them to .PID format, you said you would handle that part?

In regards to copyright etc, Monolith aren't fussed about their old games. There was a remake of their title 'Blood' not too long ago that picked up more hits than OpenClaw, but ultimately it's up to you whether you're happy to take the risk or not.

If Monolith were fussed, they're owned by WB/Time Warner and could easily find a way to take down OpenClaw as it is, but as long as it remains free and doesn't clash with any upcoming projects that they have (Eg. Claw remastered) then it should be fine.


For the smaller assets I plan on making them from scratch as well, for example crates.

It's just the Cap is hell of a lot of frames to create from scratch. 😊


Thanks,

Ryan.



________________________________

**Comment by zardav** [comment]
Any news? @iisfitblud

**Comment by Askmewho** [comment]
I think than maybe the most easy but maybe not  the most elegant way to improve the graphics its utilizing xbrz. Then ther are two options ... to create an script to transform the textures and scale all of them (maybe a copyright infringement?)...or implement xbrz inside openclaw to do it dinamicaly on the go like on emulators... but only @pjasicek will know which choice could be easier.

**Comment by farketar** [comment]
> That's great !
> 
> Here are converted images ing .PNG format:
> 
> https://drive.google.com/file/d/1l3ZgBw-HCG7CmlMrwvQQwJJQJ8FRVFq7/view
> 
> Thanks !
> 
> On a more technical side:
> 
> Currently, OpenClaw uses all of their original asset from original .REZ archive. The images in .REZ archive are stored in a proprietary .PID format (this format was only used in some of Monolith's games) and OpenClaw used .PID format directly.
> 
> **The .PID format stores 2 key informations which no normal image format does: X and Y offset of any particular image in a spritesheet.**
> 
> What that means is that if the enhanced graphics would result in for example .PNG format, I would have to rewrite some parts of my image handling, but I would ultimately take care of that.
> 
> The only thing I would require is for the new images to be X times larger than original ones where X is 2, 3 or 4 (exactly) so that image offsets could also be applied and so that images in spritesheet animations would not result in images having different offsets.

can you please resend the converted files? It seems like the link has expired

**Comment by iisfitblud** [comment]
Howdy again!

It’s been a while haha. I’m sure I could make some good progress with AI tools now. They’ve come a long way since we last spoke!

Regardless, here are the files again with a working link: https://drive.google.com/file/d/1bZpC_y_iKLooWfEGFjuABzhOgcpcIuA6/view?usp=drive_link

If these are anywhere near suitable, I can write some tools to go through and plug in all the sprites to whatever scale we like I’m fairly sure!

Thanks.

**Comment by farketar** [comment]
Hey @iisfitblud, thank you for the prompt response. Do you have some way I can convert them for myself. I was interested in level backgrounds

**Comment by iisfitblud** [comment]
If I could get a new copy of the png files or even better, show me the PID files and what I need to do to extract them, I'll use some AI tooling to get it all scaled!

**Comment by iisfitblud** [comment]
> Hey @iisfitblud, thank you for the prompt response. Do you have some way I can convert them for myself. I was interested in level backgrounds

Ah sorry. @pjasicek was working on this with me so he has the know on how to sort the files. I will certainly take a look and see what I can work out though!

**Comment by farketar** [comment]
Do you have the level backgrounds converted?

**Comment by iisfitblud** [comment]
No, sorry. I only did Captain as a POC and didn't hear anything back. As @pjasicek stated too, you can't upscale the level without upscaling every single asset either!

**Comment by farketar** [comment]
thanks @iisfitblud anyway!

**Comment by jan-kleks** [comment]
Check this out, everyone: [Broken Sword dev ‘simply couldn’t afford’ to remake it without using AI](https://www.polygon.com/23842925/broken-sword-6-bs1-remaster-charles-cecil) And this game has a cartoon style as well...

So AI definitely could help.
